addons/mrp/models/mrp_workorder.py

```diff
@@ -597,9 +597,9 @@ def button_start(self):
             vals['leave_id'] = leave.id
             return self.write(vals)
         else:
-            if self.date_planned_start > start_date:
+            if not self.date_planned_start or self.date_planned_start > start_date:
                 vals['date_planned_start'] = start_date
-                if self.duration_expected:
+                if self.duration_expected or not self.date_planned_start:
```

Comment:
I don't think you need it here too

Reply:
I think we need it here as we don't have any `date_planned_finished` if we don't have a `date_planned_start`.
So we always nee to go inside of the `if self.duration_expected or not self.date_planned_start:`  even if we don't have a `duration_expected`

Reply:
I mean do we need the `if` ? I don't see any case where we do not want to recompute `date_planned_finsihed`

Reply:
You're probably right, I'll change it.